### PR TITLE
Fix SPA syntax warnings. Fixes #472.

### LINKED ISSUE #472: Low Priority: "spa_sequence*" examples throw Syntax warning
I was running the Basal Ganglia examples using ipython notebook and I just wanted to note that there's some odd warnings that get thrown. Here's a screenshot from `spa_sequence-Context.ipynb`:

image

Is this an API problem (excessively vague warnings) or a code problem (the examples need to be fixed)?

### PR DESCRIPTION
**Pull request by jgosmann**
This is the minimal change set to fix the SPA syntax warnings by using a config.

I had to add a `similarity` function to `SPA` because the config object is only known to the `SPA` object. Thus, we have now two similarity functions (the other one in `spa.utils`). Not sure if we want to keep it that way or unify them somehow.

Also, we might add some syntactic sugar for accessing the vocabulary. Right now one has to do something like `model.config[probe.target].vocab` which does not seem very straight forward. Before it was just `probe.target.vocab`.

### CODE REVIEW

File: examples/question_control.ipynb [hunk 1]
```diff
@@ -199,9 +199,17 @@
...
      "metadata": {},
      "outputs": []
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
```
Comment (tbekolay) [inline examples/question_control.ipynb]: Extra empty cell; please remove :rainbow:

File: examples/spa_parser.ipynb [hunk 2]
```diff
@@ -204,9 +204,17 @@
...
      "metadata": {},
      "outputs": []
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
```
Comment (tbekolay) [inline examples/spa_parser.ipynb]: Extra empty cell; please remove :rainbow:

File: examples/spa_sequence.ipynb [hunk 3]
```diff
@@ -200,9 +200,17 @@
...
      "metadata": {},
      "outputs": []
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
```
Comment (tbekolay) [inline examples/spa_sequence.ipynb]: Extra empty cell; please remove :rainbow:

File: examples/spa_sequence_routed.ipynb [hunk 4]
```diff
@@ -194,9 +194,17 @@
...
      "metadata": {},
      "outputs": []
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
```
Comment (tbekolay) [inline examples/spa_sequence_routed.ipynb]: Extra empty cell; please remove :rainbow:

**Comment by tbekolay** [comment]
This LGTM! I'm going to spend a little while thinking about the `similarity` function, as that's likely to be a commonly used function; I'll either add a commit for that shortly or merge if I can't think of anything better.

**Comment by tbekolay** [comment]
Added a commit with (I think) improved function signatures for both similarity functions. See the commit message for more info. If this looks good @jgosmann and @tcstewar I'll merge!

**Comment by tcstewar** [comment]
Looks good to me!  Nice way to organize that.

**Pull request merged**